src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -480,11 +483,15 @@ private void LoadLogs(ConsoleLogsSubscription newConsoleLogsSubscription)
                         }
 
                         var logEntry = logParser.CreateLogEntry(content, isErrorOutput);
-                        if (timestampFilterDate is null || logEntry.Timestamp is null || logEntry.Timestamp > timestampFilterDate)
+                        if (logEntry.Timestamp is not null &&
+                            ((timestampFilterDate is not null && !(logEntry.Timestamp > timestampFilterDate))
+                            || PauseManager.ConsoleLogsPausedRanges.Any(range => range.IsOverlapping(logEntry.Timestamp.Value))))
```

Comment:
Can you rewrite this check to use a method instead of LINQ. Having N+1 LINQ queries might have some perf impact.